server/src/main/java/io/druid/client/DirectDruidClient.java

```diff
@@ -315,6 +322,7 @@ public InputStream nextElement()
               nodeTimeMs,
               byteCount.get() / (0.001 * nodeTimeMs) // Floating math; division by zero will yield Inf, not exception
           );
+          QueryMetrics<? super Query<T>> queryMetrics = acquireQueryMetrics();
```

Comment:
To me, calling `acquireQueryMetrics()` on three lines in a row is also misleading, because someone may think that three different objects are expected to be returned from this method. I renamed `acquireQueryMetrics()` to `acquireResponseMetrics()` and local variable accordingly.